fixed compatability with new LibGDX

EtK2000 opened the pull request:
changed return type to SnapshotArray<> as apposed to Array<>

EtK2000 committed:
fixed isses with getLifecycleListeners() 

changed return type to SnapshotArray<> as apposed to Array<>

EtK2000 committed:
woops, forgot that

EtK2000 committed:
Fixed issues with init

updated the code to make it up to date with LibGDX

EtK2000 commented:
Stating that it took my newest pull, I'll change this to the appropriate name.

intrigus commented:
It's absolutely impossible to see what you changed.

EtK2000 commented:
I know, I kinda rewrote the whole thing.
I rewrote it in Android Studio, I presume it formatted some things.

EtK2000 committed:
stupid fix

I wonder how Android Studio compiled this before...

yangweigbh commented:
hi EtK2000, thanks for your commit, can you reformat your code and submit again, i can not figure out which part has changed.

EtK2000 commented:
Sure, I'll fix it later today.
Btw, I think this breaks comparability with older LibGDX in favor of new versions.
If so, I'll check which version changed the API.

EtK2000 committed:
Formatted to be like the original

note: I only changed the core in these commits, other things might need tweaking too.

EtK2000 commented:
I have pushed my commit after formatting, now the changes are readable.
I also moved some things around to keep the order consistent between this and LibGDX.

yangweigbh commented on libgdx-cardboard-core/src/com/badlogic/gdx/backends/android/CardBoardGraphics.java:
```diff
@@ -197,22 +174,45 @@ public int getWidth () {
+    * Motorola CLIQ and the Samsung Behold II. */
    private void setupGL () {
-      if (gl20 != null) return;
+      if (config.useGL30 && glVersion.getMajorVersion() > 2) {
```
glversion is null here

EtK2000 committed:
Fix a NullPointerException

too bad, I wanted to keep it close to how AndroidGraphics works, but I guess no parameter...

EtK2000 commented:
Fixed it, thanks for pointing it out.

EtK2000 commented:
Don't forget to close issue 9.
https://github.com/yangweigbh/Libgdx-CardBoard-Extension/issues/9
